code/modules/holidays/halloween/decorators.dm

```diff
@@ -0,0 +1,98 @@
+/datum/game_decorator/halloween
+
+/datum/game_decorator/halloween/is_active_decor()
+	return (get_event_progress() != -1)
+
+/datum/game_decorator/halloween/proc/get_event_progress()
+	. = -1
+	var/cur_day = text2num(time2text(world.timeofday, "DD"))
+	var/cur_mon = text2num(time2text(world.timeofday, "MM"))
+	if(cur_mon == 10)
+		return min(0, 31 - cur_day)
+	if(cur_mon == 11 && cur_day < 4)
+		return 0
+
+/// Pumpkins decorator: adds patches of carvable/wearable pumpkins around the ground level
+/datum/game_decorator/halloween/pumpkins
+	var/pumpkin_count = 100 //! Amount of pumpkins to place
+	var/pumpkin_prob_corruption = 20
+	var/pumpkin_prob_decrease = 0.6 //! Chance reduction per day before halloween
+	var/exclusion_range = 8
+
+/datum/game_decorator/halloween/pumpkins/decorate()
+	var/list/turf/valid_turfs = list()
+	var/list/ground_levels = SSmapping.levels_by_trait(ZTRAIT_GROUND)
+	for(var/ground_z in ground_levels)
+		var/list/turf/all_turfs = block(locate(1, 1, ground_z), locate(world.maxx, world.maxy, ground_z))
+		for(var/turf/open/turf in all_turfs)
+			if(turf.is_groundmap_turf)
+				valid_turfs += turf
+
+	var/list/turf/picked_turfs = list()
```

Comment:
There are no valid_turfs counted for at least LV624 as the groundmap when this fires on localhost.

Reply:
i forgot to put /turf/open/gm (groundmap) as groundmap turf because i needed to check rivers etc, gonna add it

Reply:
should be fixed. also reduced amount of pumpkins spawned